### PR TITLE
Config fixes in clients

### PR DESCRIPTION
**Pull request by ranatrk**
#### What this PR resolves:
- Updates in clients and factories related to config

#### Changes proposed in this PR:

- make sure **j.application.JSBASEConfig** and **j.application.JSFactoryBaseClass** are used
- make sure of _init(self) in factories
- make sure of _init_new(self) in clients
- use of SCHEMATEXT
- move __jslocation__ out of init
- add _CHILDCLASS
- use of **self.[x]** instead of **self.config.data['x']**

**Fixes**: #4

### CODE REVIEW

File: Jumpscale/clients/etcd/EtcdClient.py [hunk 1]
```diff
@@ -14,34 +14,30 @@
...
-
 class EtcdClient(JSConfigClient):
-
-    def __init__(self, instance="main", data=None, parent=None, template=None, ui=None, interactive=True):
-        data = data or {}
-        super().__init__(instance=instance, data=data, parent=parent, template=_template, ui=ui, interactive=interactive)
+    _SCHEMATEXT = """
+    @url = jumpscale.etcd.client
+    host = "127.0.0.1" (S)
+    port = 2379 (I)
```
Comment (BolaNasr) [inline Jumpscale/clients/etcd/EtcdClient.py]: we need type ```ipport``` not integer

File: Jumpscale/clients/syncthing/SyncthingClient.py [hunk 2]
```diff
@@ -8,40 +8,31 @@
...
-
-    def __init__(self, instance, data={}, parent=None, interactive=False):
-        JSConfigClient.__init__(self, instance=instance,
-                                data=data, parent=parent, template=TEMPLATE, interactive=interactive)
-        c = self.config.data
+    _SCHEMATEXT = """
+    @url = jumpscale.syncthing.client
+    addr = "localhost" (S)
+    port = 0 (I)
+    sshport = 22 (I)
```
Comment (BolaNasr) [inline Jumpscale/clients/syncthing/SyncthingClient.py]: we need type ipport not integer in (port and sshport)

File: Jumpscale/clients/gitea/GiteaFactory.py [hunk 3]
```diff
@@ -15,10 +15,11 @@
...
 
 class GiteaFactory(JSConfigBase):
-
-    def __init__(self):
-        self.__jslocation__ = "j.clients.gitea"
-        JSConfigBase.__init__(self, GiteaClient)
+    __jslocation__ = "j.clients.gitea"
+    _CHILDCLASS =  GiteaClient
+    
+    # def _init(self):
```
Comment (BolaNasr) [inline Jumpscale/clients/gitea/GiteaFactory.py]: need to remove this comment line

**Review by serboctor — CHANGES_REQUESTED** [review]
Part of the cleanup is to add docstrings, tests and change anything in the syntax that doesn't follow the repo conventions for any module we work on. None of that is done in the PR. 
If it is not going to be part of this PR, are there tasks created specifically for these modules?

**Comment by BolaNasr** [comment]
@serboctor 
This PR isn't for cleaning up it just makes the clients work with the new config manager , after we are sure all clients works Good , we will cleanup all .

File: Jumpscale/clients/s3/S3Client.py [hunk 4]
```diff
@@ -15,27 +15,24 @@
...
     """
-
-    def __init__(self, instance, data={}, parent=None, interactive=False):
-        JSConfigBase.__init__(self, instance=instance,
-                              data=data, parent=parent, template=TEMPLATE, interactive=interactive)
-        c = self.config.data
+    _SCHEMATEXT = """
+    @url = jumpscale.s3.client
+    address = "" (S)
+    port = 9000 (I)
```
Comment (BolaNasr) [inline Jumpscale/clients/s3/S3Client.py]: should add type ipport not int

File: Jumpscale/clients/sshagent/SSHAgent.py [hunk 5]
```diff
@@ -14,41 +18,41 @@ def keyname_default(self):
...
+        j.sal.fs.chmod(self.path, 0o600)
+        if self.passphrase:
             self._logger.debug("load with passphrase")
             C = """
                 echo "exec cat" > ap-cat.sh
                 chmod a+x ap-cat.sh
                 export DISPLAY=1
-                echo {passphrase} | SSH_ASKPASS=./ap-cat.sh ssh-add -t {duration} {path}
-                """.format(path=path0, passphrase=passphrase, duration=duration)
+                echo {self.passphrase} | SSH_ASKPASS=./ap-cat.sh ssh-add -t {duration} {path}
```
Comment (BolaNasr) [inline Jumpscale/clients/sshagent/SSHAgent.py:55]: change path to self.path

**Review by BolaNasr — COMMENTED** [review]
@ranatrk  please fix conflict in files

File: Jumpscale/clients/intercom/IntercomClient.py [hunk 6]
```diff
@@ -101,4 +88,4 @@ def get_admin(self, name):
         for admin in admins:
             if admin.name == name:
                 return admin
-        return None
+        return None
```
Comment (serboctor) [inline Jumpscale/clients/intercom/IntercomClient.py]: No new line at the end of the file

File: Jumpscale/clients/postgresql/PostgresqlClient.py [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+            "postgresql://%(login)s:%(passwd)s@%(ipaddr)s:%(port)s/%(dbname)s" % self.__dict__)
+
+        # reflect the models
+        Base.prepare(engine, reflect=True)
+
+        session = Session(engine)
+
+        return Base, session
+
+    def peewee_client_get(self):
```
Comment (serboctor) [inline Jumpscale/clients/postgresql/PostgresqlClient.py]: if this isnt referenced anywhere, can you remove it?

File: Jumpscale/clients/postgresql/PostgresqlFactory.py [hunk 8]
```diff
@@ -16,27 +9,31 @@
...
-    def __init__(self):
-        self.__jslocation__ = "j.clients.postgres"
+    def _init(self):
         self.__imports__ = "sqlalchemy"
-        JSConfigFactory.__init__(self, PostgresClient)
 
-    def createdb(self, db, ipaddr="localhost", port=5432, login="postgres", passwd="rooter"):
+    def db_create(self, db, ipaddr="localhost", port=5432, login="postgres", passwd="rooter"):
+        """Create new database
+        @param db: db name to be created
```
Comment (serboctor) [inline Jumpscale/clients/postgresql/PostgresqlFactory.py]: these docs dont match the one in the contribution guidelines, we use `:` not `@`. Check the example in the readme and modify this and the docs below too, please.

File: Jumpscale/clients/zdb/ZDBAdminClient.py [hunk 9]
```diff
@@ -1,23 +1,25 @@
...
                     https://github.com/rivine/0-db/blob/master/README.md
         """
-        ZDBClientBase.__init__(self, addr=addr, port=port, mode=mode, secret=secret,admin=True)
+        ZDBClientBase._init(self)
+        self.admin = True
         self._system = None
         # self._logger_enable()
         if self.secret:
-            #authentication should only happen in zdbadmin client
+            import ipdb
```
Comment (serboctor) [inline Jumpscale/clients/zdb/ZDBAdminClient.py]: remove this please

File: Jumpscale/sal/hostfile/HostFile.py [hunk 10]
```diff
@@ -80,13 +91,4 @@ def hostnames_set(self, ip, hostnames):
                 search_obj.group(0), '%s %s\n' % (ip, hostnames))
         else:
             filecontents += '%s %s\n' % (ip, hostnames)
-
```
Comment (serboctor) [inline Jumpscale/sal/hostfile/HostFile.py]: this shouldnt be removed...I thinked you messed up something while merging

**Review by serboctor — CHANGES_REQUESTED** [review]

**Review by serboctor — APPROVED** [review]

**Pull request merged**